dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/telnet/InvokeTelnetHandler.java

```diff
@@ -150,6 +163,17 @@ public String telnet(Channel channel, String message) {
         return buf.toString();
     }
 
+    private Invoker<?> findInvoker(ProviderModel provider) {
+        for (Exporter<?> exporter : DubboProtocol.getDubboProtocol().getExporters()) {
+            Invoker<?> invoker = exporter.getInvoker();
```

Comment:
I think the getExporters() method should move to the AbstractProtocol.java. So we can process non-dubbo protocol invokers. Or this may cause null pointer exception if there is no DubboProtocol instance

Reply:
@tswstarplanet 
`InvokeTelnetHandler` is actually defined in module `dubbo-rpc-dubbo`, so it's always used for dubbo protocol invokers, it's not implemented in other protocols.
But it's a good point, maybe we shouldn't just use `ApplicationModel#allProviderModels` to list services, we also need to filter out non-dubbo protocol invokers.

Reply:
@tswstarplanet I added hint when no dubbo protocol invoker found, I thinks it's enough. After all it's mainly for debugging purposes.